packages/next/src/build/webpack/loaders/lightningcss-loader/src/loader.ts

```diff
@@ -0,0 +1,195 @@
+import valueParser from 'next/dist/compiled/postcss-value-parser'
+import type { LoaderContext } from 'webpack'
+import type { ILightningCssLoaderConfig, VisitorOptions } from './interface'
+import { ECacheKey } from './interface'
+import { transform as transformCss, type Visitor } from 'lightningcss'
+import { Buffer } from 'buffer'
+import { getTargets } from './utils'
+import path from 'path'
+import {
+  getImportCode,
+  type ApiParam,
+  type ApiReplacement,
+  type CssExport,
+  type CssImport,
+  getModuleCode,
+  getExportCode,
+} from './codegen'
+import {
+  getFilter,
+  getPreRequester,
+  isUrlRequestable,
+} from '../../css-loader/src/utils'
+import { stringifyRequest } from '../../../stringify-request'
+
+function createVisitor(
+  options: ILightningCssLoaderConfig,
+  visitorOptions: VisitorOptions,
+  apis: ApiParam[],
+  imports: CssImport[],
+  replacements: ApiReplacement[]
+): Visitor<{}> {
+  const importUrlToNameMap = new Map<string, string>()
+
+  return {
+    Rule(node) {
```

Comment:
FYI doing 

```js
Rule: {
  Import(node) {
    // ...
  }
}
```

should be more efficient because it will only call into JS for `@import` rules rather than needing to serialize and call JS for all rules.

Reply:
I need to change the import rule to something different

Reply:
yeah you can return a different type of rule as well, or `[]` to remove the rule, or even multiple rules. [Example](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%40import%20%5C%22foo.css%5C%22%3B%5Cn%5Cn%22%2C%22visitorEnabled%22%3Atrue%2C%22visitor%22%3A%22%7B%5Cn%20%20Rule%3A%20%7B%5Cn%20%20%20%20import(rule)%20%7B%5Cn%20%20%20%20%20%20return%20%7B%5Cn%20%20%20%20%20%20%20%20type%3A%20'style'%2C%5Cn%20%20%20%20%20%20%20%20value%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20selectors%3A%20%5B%5B%7Btype%3A%20'class'%2C%20name%3A%20'hi'%7D%5D%5D%2C%5Cn%20%20%20%20%20%20%20%20%20%20declarations%3A%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20declarations%3A%20%5B%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7Bproperty%3A%20'color'%2C%20value%3A%20%7Btype%3A%20'rgb'%2C%20r%3A%202%2C%20g%3A%202%2C%20b%3A%202%2C%20alpha%3A%201%7D%7D%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%5D%5Cn%20%20%20%20%20%20%20%20%20%20%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20loc%3A%20rule.value.loc%5Cn%20%20%20%20%20%20%20%20%7D%5Cn%20%20%20%20%20%20%7D%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)

Reply:
Ah... it was `Rule.import`. Thank you!